contrib/epee/include/console_handler.h

```diff
@@ -489,24 +420,50 @@ namespace epee
 #endif
     }
 
-    /// Throws std::out_of_range on bad command with what() set to the command name, otherwise
+    /// Throws invalid_command on bad command with what() set to the command name, otherwise
     /// returns the result of the command (true generally means success, false means failure).
+    struct invalid_command : std::invalid_argument { using std::invalid_argument::invalid_argument; };
     bool process_command(const std::vector<std::string>& cmd)
     {
       if(!cmd.size())
-        throw std::out_of_range{"(empty)"};
+        throw invalid_command{"(empty)"};
       auto it = m_command_handlers.find(cmd.front());
       if (it == m_command_handlers.end())
-        throw std::out_of_range{cmd.front()};
+        throw invalid_command{cmd.front()};
       return it->second.first(std::vector<std::string>{cmd.begin()+1, cmd.end()});
     }
 
-    bool process_command(const std::string& cmd)
+    bool process_command_and_log(const std::vector<std::string> &cmd)
+    {
+      try
+      {
+        return process_command(cmd);
+      }
+      catch (const invalid_command &e)
+      {
+#ifdef HAVE_READLINE
+        rdln::suspend_readline pause_readline;
+#endif
```

Comment:
These `ifdef` all over for readline are ugly.  #1159 solves that (by making `rdln::suspend_readline` a compilable no-op when we don't have readline support).